src/web/ui/Chrome.js

```diff
@@ -474,21 +471,13 @@ exports = Class(CenterLayout, function (supr) {
     };
   }
 
-  this._setViewport = function (size, viewport, scale) {
+  this._setViewport = function (size, viewport, scale, dpr) {
```

Comment:
Would it be better to get dpr from this.getDevicePixelRatio() instead of passing it in to _setViewport? That way other calls to _setViewport won't need to handle the DPR as well. Additionally, it could lead to confusion if the dpr used in _setViewport didn't match what is returned from getDevicePixelRatio.